public/pages/workflow_detail/workflow_detail.tsx

```diff
@@ -64,21 +58,28 @@ export function WorkflowDetail(props: WorkflowDetailProps) {
   const workflow = workflows[workflowId];
   const workflowName = workflow ? workflow.name : DEFAULT_NEW_WORKFLOW_NAME;
 
+  const {
+    chrome: { setBreadcrumbs },
+  } = getCore();
+  const uiSettings = getUISettings();
+  const showActionsInHeader = uiSettings.get('home:useNewHomePage');
   useEffect(() => {
-    if (dataSourceEnabled) {
-      getCore().chrome.setBreadcrumbs([
-        BREADCRUMBS.FLOW_FRAMEWORK,
-        BREADCRUMBS.WORKFLOWS(dataSourceId),
-        { text: workflowName },
+    if (showActionsInHeader) {
+      setBreadcrumbs([
+        BREADCRUMBS.TITLE_WITH_REF(
+          dataSourceEnabled ? dataSourceId : undefined
+        ),
+        BREADCRUMBS.WORKFLOW_NAME(workflowName),
+        { text: '' },
       ]);
     } else {
-      getCore().chrome.setBreadcrumbs([
+      setBreadcrumbs([
         BREADCRUMBS.FLOW_FRAMEWORK,
-        BREADCRUMBS.WORKFLOWS(),
+        BREADCRUMBS.WORKFLOWS(dataSourceEnabled ? dataSourceId : undefined),
         { text: workflowName },
       ]);
     }
```

Comment:
even better, I think this could be combined into just a single `setBreadcrumbs` call:
```
      setBreadcrumbs( showActionsInHeader ? [
        BREADCRUMBS.TITLE_WITH_REF(
          dataSourceEnabled ? dataSourceId : undefined
        ),
        BREADCRUMBS.WORKFLOW_NAME(workflowName),
        { text: '' },
      ] : [
        BREADCRUMBS.FLOW_FRAMEWORK,
        BREADCRUMBS.WORKFLOWS(dataSourceEnabled ? dataSourceId : undefined),
        { text: workflowName },
      ]);
```